pkg/cache/application.go

```diff
@@ -511,11 +517,24 @@ func (app *Application) handleRejectApplicationEvent(event *fsm.Event) {
 		fmt.Sprintf("application %s is rejected by scheduler", app.applicationID)))
 }
 
+func placeholderCleanup(app *Application) {
+	placeholderManager := getPlaceholderManager()
+	if placeholderManager != nil {
+		placeholderManager.cleanUp(app)
+	}
+}
+
```

Comment:
`getPlaceholderManager()` cannot return a nil we can simplify this by directly calling this inside the go function while handling the events:
```
go func() {
    getPlaceholderManager().cleanup(app)
}
```